src/Scanner.js

```diff
@@ -246,12 +252,16 @@ class DefaultScanner {
 
             if (element.parentElement) {
                 element = element.parentElement;
-                const inputs = element.querySelectorAll(this.matching.cssSelector('formInputsSelector'));
-                const buttons = element.querySelectorAll(this.matching.cssSelector('submitButtonSelector'));
-                // If we find a button or another input, we assume that's our form
-                if (inputs.length > 1 || buttons.length) {
-                    // found related input, return common ancestor
-                    return element;
+                // check if current element is the only child of the parent, and increase traversal only then
```

Comment:
Make this more explicit. Your comment is repeating verbatim what the code does.

```suggestion
                // If the parent is a redundant component (only contains a single element or is a shadowRoot) do not increase the traversal count.
                // This way we can safely traverse more meaningful layers and increase accuracy.
```